package-rust.py

```diff
@@ -176,15 +180,26 @@ def run(args):
 if channel == "stable":
     CFG_RELEASE=CFG_RELEASE_NUM
     CFG_PACKAGE_VERS=CFG_RELEASE_NUM
+    # UpgradeCode shoud stay the same for all MSI versions in channel
+    CFG_UPGRADE_CODE="1C7CADA5-D117-43F8-A356-DF15F9FBEFF6"
+    CFG_MSI_VERSION=CFG_RELEASE_NUM
 elif channel == "beta":
     CFG_RELEASE=CFG_RELEASE_NUM + "-beta" + CFG_BETA_CYCLE
     CFG_PACKAGE_VERS="beta"
+    CFG_UPGRADE_CODE="5229EAC1-AB7C-4A62-9881-6FAD2DE7D0F9"
+    CFG_MSI_VERSION=CFG_RELEASE_NUM + "." + CFG_BETA_CYCLE
 elif channel == "nightly":
     CFG_RELEASE=CFG_RELEASE_NUM + "-nightly"
     CFG_PACKAGE_VERS="nightly"
+    CFG_UPGRADE_CODE="B94FF1C2-2C7B-4859-A08B-546815516FDA"
+    now=datetime.datetime.now()
+    build=now.year*10+now.month*100+now.day
+    CFG_MSI_VERSION=CFG_RELEASE_NUM+"."+str(build)
 elif channel == "dev":
     CFG_RELEASE=CFG_RELEASE_NUM + "-dev"
     CFG_PACKAGE_VERS=CFG_RELEASE_NUM + "-dev"
+    CFG_UPGRADE_CODE="7E6D1349-2773-4792-B8CD-EA2685D86A99"
+    CFG_MSI_VERSION="255.255.65535.99999"
```

Comment:
FYI all this logic of pulling stuff out of the source code will change when we [start dumping the version info into the binary tarballs](https://github.com/rust-lang/rust/pull/20388). I don't think that should affect anything here, besides the necessary rewrite.